xarray/core/nputils.py

```diff
@@ -79,3 +79,59 @@ def array_ne(self, other):
     with warnings.catch_warnings():
         warnings.filterwarnings('ignore', r'elementwise comparison failed')
         return _ensure_bool_is_ndarray(self != other, self, other)
+
+
+def _is_contiguous(positions):
+    """Given a non-empty list, does it consist of contiguous integers?"""
+    previous = positions[0]
+    for current in positions[1:]:
+        if current != previous + 1:
+            return False
+        previous = current
+    return True
+
+
+def _advanced_indexer_subspaces(key):
+    """Indices of the advanced indexes subspaces for mixed indexing and vindex.
+    """
+    if not isinstance(key, tuple):
+        key = (key,)
+    advanced_index_positions = [i for i, k in enumerate(key)
+                                if not isinstance(k, slice)]
+
+    if (not advanced_index_positions or
+            not _is_contiguous(advanced_index_positions)):
+        # Nothing to reorder: dimensions on the indexing result are already
+        # ordered like vindex. See NumPy's rule for "Combining advanced and
+        # basic indexing":
+        # https://docs.scipy.org/doc/numpy/reference/arrays.indexing.html#combining-advanced-and-basic-indexing
+        return (), ()
+
+    non_slices = [k for k in key if not isinstance(k, slice)]
+    if len(non_slices) == 1:  # older np.broadcast does not support one array
```

Comment:
Is this still true, even with requiring numpy 1.11?

Reply:
No. As you pointed out we no longer need this with numpy=1.11. Removed.